src/App.tsx

```diff
@@ -49,39 +47,23 @@ function App() {
     setChecktedItems(not(checkedItems, leftCheckedItems));
   }
 
-  function handleInputChange(e) {
-    setInputValue(e.target.value);
-  }
-
-  function handleAddToList() {
-    setFirstListItems([...firstListItems, inputValue]);
-    setInputValue("");
+  function handleAddToList(item: string) {
+    setFirstListItems((prev) => [...prev, { name: item, id: crypto.randomUUID()}]);
   }
 
-  const debouncedCalcWordCount = debounce(
-    (text) => {
-      const wordCount = calculateWordCount(text)
-      setCurrWordCount(wordCount)
-    }, 100)
-
-  useEffect(
-    function () {
-      debouncedCalcWordCount(inputValue)
-    },
-    [inputValue],
-  );
-
```

Comment:
This is not needed at all. When you are setting state inside useEffect, please think twice. Most likely you can avoid it. It is considered as bad practice. 

To resolve it, I just used the calculater function inside the template. It automatically did the job.